### ISSUE TITLE
client: CLI application

### ISSUE THREAD

**Issue by chappjc** [decred/dcrdex#38]
In addition to the [Core Client Components](https://github.com/decred/dcrdex/milestone/3) milesone issues, a command line interface (CLI) application needs to be created for the DEX client.

Work on this issue should **follow** development and testing of the core client components.

---

A suggestion I have for making the CLI app as interactive and intuitive as possible is to make use of **menus and a clear workflow**.  To this goal, one tool that I have had good luck with in dcrdata's `democlient` CLI for the `psclient` package is https://github.com/AlecAivazis/survey.

**Comment by chappjc** [comment]
companyzero/zkc has some good UI work too.  https://github.com/companyzero/zkc/tree/master/zkclient

**Comment by devwarrior777** [comment]
Did you look at the grin CLI also?

**Comment by chappjc** [comment]
@devwarrior777 No, would you link to it?  I'm not sure which app you're referring to.

**Comment by devwarrior777** [comment]
https://www.coingecko.com/buzz/how-to-use-grin-wallet-to-send-receive-grin-coins

scroll down about half way for B/W CLI

**Comment by JoeGruffins** [comment]
I would really like to work on this. survey looks good to me. 
The dep tree:
<details close>
<summary>deps</summary>

```
github.com/AlecAivazis/survey/v2
 |-bytes
  |-internal/bytealg
   |-internal/cpu
   |-unsafe
  |-unicode
  |-unicode/utf8
 |-errors
  |-internal/reflectlite
 |-fmt
  |-internal/fmtsort
   |-sort
  |-math
   |-math/bits
  |-strconv
  |-sync
   |-internal/race
   |-sync/atomic
 |-github.com/AlecAivazis/survey/v2/core
  |-github.com/mgutz/ansi
   |-github.com/mattn/go-colorable
    |-github.com/mattn/go-isatty
     |-golang.org/x/sys/unix
      |-encoding/binary
      |-net
       |-C
       |-context
       |-vendor/golang.org/x/net/dns/dnsmessage
       |-internal/nettrace
       |-internal/poll
       |-internal/singleflight
       |-math/rand
      |-syscall
       |-internal/oserror
  |-text/template
   |-net/url
   |-path/filepath
   |-text/template/parse
  |-time
 |-github.com/AlecAivazis/survey/v2/terminal
  |-bufio
 |-github.com/kballard/go-shellquote
 |-io
 |-io/ioutil
 |-os
  |-internal/syscall/unix
  |-internal/testlog
 |-os/exec
 |-reflect
 |-regexp
  |-regexp/syntax
 |-runtime
  |-runtime/internal/atomic
  |-runtime/internal/math
  |-runtime/internal/sys
 |-strings
```
</details>

**Comment by chappjc** [comment]
@JoeGruffins I actually made another issue for the UI: https://github.com/decred/dcrdex/issues/80

survey is nice and I've used it before for prompts, but it won't do all we need.  The screen will need to be subdivided to be most useful.  @devwarrior777 linked to another app's CLI, and the design is a good example.
![image](https://user-images.githubusercontent.com/9373513/69435984-2500dd80-0d06-11ea-8bf8-3f09d1f5d641.png)

We have very different needs, but the menu design there is reasonable.

**Comment by chappjc** [comment]
To start, we should just try to get menus and actions worked out, and not worry about order tables and charts.  For the book, one of the menu items could simply be to show top orders on either side of the market.   But it would be nice to have some kind of reasonably nice UI.

**Comment by buck54321** [comment]
Agreed. I'm hesitant to start this before persistent storage and order communications are done, but if @JoeGruffins is itching to do it, that's okay. As @chappjc implied, let's not worry about integrating the DEX parts at first. Step 1 is to figure out a good UI for the menus and sections, and distill it down to a clean internal API that we can build on.

Let's also keep the work in perspective. This app will see very limited use and our time is better used developing the underlying components.

**Comment by JoeGruffins** [comment]
Understood. I'll see if I can get some idea of what's best and do a little research.  I'll post back with what I find.

**Comment by buck54321** [comment]
This issue is resolved by @JoeGruffins work on **dexcctl** and **rpcserver**. Obviously more work to do, but can we close this issue?

**Comment by chappjc** [comment]
Yeah, agreed.  Resolved by #181
